clean/new_orleans_pd_stop_and_search.py

```diff
@@ -0,0 +1,250 @@
+        .str.extract(r'(?:(consent form completed\: (\w+)))')
+
+    df.loc[:, 'consent_form_completed'] = col[0]\
+        .str.replace(r'consent form completed\: ', '', regex=True)
+    return df
+
+
+def extract_body_search_column(df):
+    col = df.actionstaken.str.lower().str.strip()\
+        .str.extract(r'(?:(stripbody cavity search\: (\w+)))')
+
+    df.loc[:, 'strip_body_cavity_search'] = col[0]\
+        .str.replace(r'stripbody cavity search\: ', '', regex=True)
+    return df
+
+
+def extract_exit_vehicle_column(df):
+    col = df.actionstaken.str.lower().str.strip()\
+        .str.extract(r'(?:(exit vehicle\: (\w+)))')
+
+    df.loc[:, 'exit_vehicle'] = col[0]\
+        .str.replace(r'exit vehicle\: ', '', regex=True)
+    return df.drop(columns='actionstaken')
+
+
+def clean_assigned_department(df):
+    df.loc[:, 'assigned_department'] = df.officerassignment.str.lower().str.strip()\
+        .str.replace('pib', 'public integrity bureau', regex=False)\
+        .str.replace('fob', 'field operations bureau', regex=False)\
+        .str.replace('msb', 'management services bureau', regex=False)\
+        .str.replace('superintendent', 'office of the superintendent', regex=False)\
+        .str.replace('isb', 'investigations and support bureau', regex=False)\
+        .str.replace(r'((.+) district)', '', regex=True)
+    return df.drop(columns='officerassignment')
+
+
+def consolidate_name_and_badge_columns(df):
+    df.loc[:, 'officer1badgenumber'] = df.officer1badgenumber.astype(str)\
+        .str.replace(r'^(\w+)$', r'(\1)', regex=True)
+    df.loc[:, 'officer2badgenumber'] = df.officer2badgenumber.astype(str)\
+        .str.replace(r'^(\w+)\.(\w+)$', r'(\1)', regex=True)
+
+    df.loc[:, 'officer1name'] = df.officer1name.str.cat(df.officer1badgenumber, sep=' ')
+    df.loc[:, 'officer2name'] = df.officer2name.str.cat(df.officer2badgenumber, sep=' ')
+
+    df.loc[:, 'officer_names_and_badges'] = df.officer1name.str.cat(df.officer2name, sep='/')
+    return df
+
+
+def split_rows_with_multiple_officers(df):
+    i = 0
+    for idx in df[df.officer_names_and_badges.fillna('').str.contains('/')].index:
+        s = df.loc[idx + i, 'officer_names_and_badges']
+        parts = re.split(r"\s*(?:\/)\s*", s)
+        df = duplicate_row(df, idx + i, len(parts))
+        for j, name in enumerate(parts):
+            df.loc[idx + i + j, 'officer_names_and_badges'] = name
+        i += len(parts) - 1
+    return df
+
+
+def extract_badge_number(df):
+    badges = df.officer_names_and_badges.str.extract(r'\(\w+)\)')
+    df.loc[:, 'badge_number'] = badges[0]\
+        .str.replace(r'\(|\)', '', regex=True)
+    return df
+
+
+def extract_rank_desc(df):
+    return df
+
+
+def split_names(df):
+    return df
+
+
+def clean():
+    df = pd.read_csv(data_file_path('raw/ipm/new_orleans_pd_stop_and_search_2007_2021.csv'))\
+        .pipe(clean_column_names)\
+        .drop(columns=['subjectage'])\
```

Comment:
The raw file seems to be the wrong name? Also when I changed the name it shows me this error: `KeyError: "['subjectage'] not found in axis"`. Any idea? Also curious how the file name can be wrong in the first place. Are you not using `scripts/rawfiles.sh` to get the raw file name perhaps?

Reply:
Sorry. I added a version that I cleaned for some initial analysis to the `raw_datasets.json`. I just replaced it with the raw version.